### DISCUSSION TITLE
borg extract: Date/time on folders/directories show date of extract vs. original date/times at backup - expected?

### DISCUSSION THREAD

**Discussion by rxgh99** [borgbackup/borg#7298] [Q&A] [answered]
Hi, 

On testing Vorta/Borg backup/restore, I've noticed that the extract process restores directories with the date of when the extract was performed vs. the original date of the directory structure. Original dates/times are preserved on files, but not folders. Is this expected behavior? (My peference is for full restore of a dir structure to reflect original dates/times when the backup occurred, hence this question.)

Thanks.

**Comment by ThomasWaldmann** [comment, marked as answer]
mtime should be correct. If it is not, file a bug.

With atime, it is a bit tricky, it will depend on whether the OS supports/permits opening stuff without touching atime. Guess it should work on Linux as root.

ctime is something borg can not influence, this is only controlled by the kernel and thus will be the current datetime.

**Reply by rxgh99** [reply]
Ok, thanks. 

Current insights suggest mtime is not preserved -- I'll finalize some testing/checking here (on macOS) and will look to raise a bug.

**Reply by ThomasWaldmann** [reply]
https://github.com/borgbackup/borg/pull/7299/files I added a unit test for the extracted directory mtime.

It worked ok for me.

So guess I'll need a small script from you that reproduces the issue. Also please give the precise borg version you use.

**Reply by rxgh99** [reply]
Ok - interesting. I will do some more testing and reply back here. 

Currently using v1.2.3 CLI (from Brew) and with Vorta (0.8.10) on macOS 12.6.2 (Monterey).

**Reply by rxgh99** [reply]
I've learned a few things through some testing here and I don't think this is a bug :) Apologies for any suggestion there; I think I conflated a couple of expectations during some restore testing scenarios. But it has also made me re-consider my backup approach and strategy based on my learnings and am considering a potential discussion on Vorta sources selection approach :) 

Context: 
I was running several restore scenarios: single file, multiple files across different directory structures, multiple directories, full volume restore. 

Across all of these scenarios, I likely conflated certain expectations; for example, during a single-file restore I observed the times on restored directories to be the restore date and I mistakenly expected the directory folders to have the same mtime, etc. as the original backup archive. Instead, only the item selected for restore has the restored time(s).  

In addition, I realised that on restore, only directories _underneath the source root_ will have their times restored per the archive being restored from; any directories created that are part of the source path will have dates of restore vs. date captured in the archive. 

So this has made me re-evaluate how I'm capturing the source directories and that's leading me to consideration of higher-level source path supported by exclusion rules, etc. I'm primarily running Vorta at present, so have some thoughts on potential options there to help facilitate easier inclusion selection and exclusion by inference.

All that to say - no bug! But I've found it useful to understand in more detail the behaviors and implications of source paths in restore as well as restore execution itself.
